draft-jholland-quic-multicast.md

```diff
@@ -64,15 +64,24 @@ This document does not define any multicast transport except server to client.
 
 # Multicast Session
 
-A multicast session (or just session) is a one-way communication stream by which a server can transport data to a client as part of the QUIC connection between the server and client.
+A multicast session (or just session) is a one-way network path that a server can use as an alternate path to send QUIC connection data to a client.
 
-Multicast sessions are additional network paths that can optionally be used by the server and client to transport application data concurrently with other network paths used by the connection.
+Multicast sessions are designed to leverage multicast IP and to be shared by many different connections simultaneously for unidirectional server-initiated data.
+Many servers can use the same session to send the same data to many clients.
 
-Sessions are designed to leverage multicast IP, so all clients joined to a session will receive identical data packets for that session at the IP layer.
-Sessions are designed to be shared by many connections so that many servers can use the same session to communicate the same data to many clients.
+QUIC connections are defined in Section 5 of {{RFC9000}} and are not changed in this document; each connection is a shared state between a client and a server.
 
-Note that each QUIC connection still is defined only between a single client and a single server.
-But when using the multicast extension defined in this document, some of the data transmitted from server to client may be data that used a network transport path that was shared with other connections.
+Sessions carry only 1-RTT packets.
+Packets associated with a session contain a Session ID in place of a Destination Connection ID.
+(A Session ID cannot be zero length.)
+This adds a layer of indirection to the process described in Section 5.2 of {{RFC9000}}} for matching packets to connections upon receipt.
+Incoming packets received on the network path associated with a session use the Session ID to associate the packet with a joined session.
```

Comment:
How can a client differentiate if a packet belongs to a session or a connection if both use the same ID space? There should probably be a mechanic to force clients to not use some connection IDs for unicast connections. I think only doing it in the MC_SESSION_PROPERTIES frame is too late, it should probably be done immediately after the handshake of the unicast connection. Its basically a frame telling the client "Do not use any of these connection IDs for your unicast streams as we might have a multicast session that uses this ID. If you (by sheer bad luck) already use one of them for a unicast connection (i.e. the initially created one), issue a new connection ID and retire the old one immediately."

Reply:
That is a very good idea on a problem that has been bugging me.  I was thinking to do it by the receive path, but I like your idea better.

I guess session properties should implicitly reserve the session id, but also a MC_RESERVE_SESSIONIDS frame to reserve a list of IDs without sending their properties.

I think if there is a collision it doesn't matter until the server issues a JOIN, but if the client has not yet migrated connection ID it can refuse the join with an "ID Collision" reason and we don't really need other signaling about it I think?  Actually it occurs to me the server should already know if the connection ID is in use, and can determine when it has closed to retry the join?

Reply:
Made a new issue (#9) for this point, that'll be a separate PR.